src/internal_macros.rs

```diff
@@ -45,41 +45,6 @@ macro_rules! impl_consensus_encoding {
 
 macro_rules! impl_array_newtype {
     ($thing:ident, $ty:ty, $len:expr) => {
-        impl $thing {
-            #[inline]
-            /// Converts the object to a raw pointer
-            pub fn as_ptr(&self) -> *const $ty {
-                let &$thing(ref dat) = self;
-                dat.as_ptr()
-            }
-
-            #[inline]
-            /// Converts the object to a mutable raw pointer
-            pub fn as_mut_ptr(&mut self) -> *mut $ty {
-                let &mut $thing(ref mut dat) = self;
-                dat.as_mut_ptr()
-            }
-
-            #[inline]
-            /// Returns the length of the object as an array
-            pub fn len(&self) -> usize { $len }
-
-            #[inline]
-            /// Returns whether the object, as an array, is empty. Always false.
-            pub fn is_empty(&self) -> bool { false }
-
-            #[inline]
-            /// Returns the underlying bytes.
-            pub fn as_bytes(&self) -> &[$ty; $len] { &self.0 }
-
-            #[inline]
-            /// Returns the underlying bytes.
-            pub fn to_bytes(&self) -> [$ty; $len] { self.0.clone() }
-
-            #[inline]
-            /// Returns the underlying bytes.
-            pub fn into_bytes(self) -> [$ty; $len] { self.0 }
```

Comment:
I don't seem to see where these last 5 methods have gone. Especially the last 3 are really useful.

Reply:
Really? because you basically get them for free with the `Index` impl.
but if you think they're useful i'll keep them

Reply:
Hmm, true that you get them with `Index`. The only one would be `into_inner`, not sure in what cases one wants to keep the inner value, but I think we ought support it, no?

Reply:
Did you add the into_ method? I think `into_inner` is a useful method always. Any newtype ever probably should provide it without any overhead like reallocation.

Reply:
Reverted that change